src/ccs/lcccs.rs

```diff
@@ -23,51 +23,60 @@ use core::{cmp::max, marker::PhantomData};
 use ff::{Field, PrimeField};
 use flate2::{write::ZlibEncoder, Compression};
 use itertools::concat;
+use rand_core::RngCore;
 use rayon::prelude::*;
 use serde::{Deserialize, Serialize};
 use sha3::{Digest, Sha3_256};
 use std::ops::{Add, Mul};
 use std::sync::Arc;
 
 /// A type that holds a LCCCS instance
-#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+#[derive(Clone, Debug, Default, PartialEq, Eq, Serialize, Deserialize)]
 #[serde(bound = "")]
 pub struct LCCCS<G: Group> {
-  pub(crate) C: Commitment<G>,
-  pub(crate) x: Vec<G::Scalar>,
-  pub(crate) u: G::Scalar,
+  pub(crate) w_comm: Commitment<G>,
   pub(crate) v: Vec<G::Scalar>,
   // Random evaluation point for the v_i
-  pub r_x: Vec<G::Scalar>,
-  // This should not need to be here. Should be a reference only.
-  pub(crate) matrix_mles: Vec<MultilinearPolynomial<G::Scalar>>,
-  pub(crate) ccs: CCSShape<G>,
+  pub(crate) r_x: Vec<G::Scalar>,
+  pub(crate) z: Vec<G::Scalar>,
 }
 
 impl<G: Group> LCCCS<G> {
-  // XXX: Double check that this is indeed correct.
-  /// Samples public parameters for the specified number of constraints and variables in an CCS
-  pub fn commitment_key(&self) -> CommitmentKey<G> {
-    let total_nz = self.ccs.M.iter().fold(0, |acc, m| acc + m.coeffs().len());
-
-    G::CE::setup(b"ck", max(max(self.ccs.m, self.ccs.t), total_nz))
+  /// Generates a new LCCCS instance from a given randomness, CommitmentKey & witness input vector.
+  /// This should only be used to probably test or setup the initial NIMFS instance.
```

Comment:
TODO to move out as test function

Reply:
If I do so, I can't do `LCCCS::new`. I think it's better to leave it like this.
Added `#[cfg(test)]` for it.

Reply:
Fixed in 52f6a62

Reply:
Just realized I can't do `#[cfg(test)]` as this function is used by the NIMFS.
That's why it was marked as `pub(crate)`. So that it allows for unit testing setup + internal crate fn calls.